vocode/streaming/streaming_conversation.py

```diff
@@ -806,28 +809,30 @@ def warmup_synthesizer(self):
     def mark_last_action_timestamp(self):
         self.last_action_timestamp = time.time()
 
-    def broadcast_interrupt(self):
+    async def broadcast_interrupt(self):
         """Stops all inflight events and cancels all workers that are sending output
 
         Returns true if any events were interrupted - which is used as a flag for the agent (is_interrupt)
         """
-        num_interrupts = 0
-        while True:
-            try:
-                interruptible_event = self.interruptible_events.get_nowait()
-                if not interruptible_event.is_interrupted():
-                    if interruptible_event.interrupt():
-                        logger.debug(
-                            f"Interrupting event {type(interruptible_event.payload)} {interruptible_event.payload}",
-                        )
-                        num_interrupts += 1
-            except queue.Empty:
-                break
-        self.agent.cancel_current_task()
-        self.agent_responses_worker.cancel_current_task()
-        if self.actions_worker:
-            self.actions_worker.cancel_current_task()
-        return num_interrupts > 0
+        async with self.interrupt_lock:
```

Comment:
it isn't to prevent simultaneous interrupts - it's to synchronize this bit of code with how we send chunks to playback, see my explanation below.

we only call `broadcast_interrupt` in two places - when we receive transcriptions / when we terminate the conversation. these two bits can probably mess with each other (but they have always messed with each other), but is OK-ish since the conversation ends